pkg/apis/stork/v1alpha1/types.go

```diff
@@ -212,3 +217,193 @@ type MigrationList struct {
 
 	Items []Migration `json:"items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StorageCluster represents a storage cluster
+type StorageCluster struct {
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+	Spec            StorageClusterSpec   `json:"spec"`
+	Status          StorageClusterStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StorageClusterList is a list of StorageCluster
+type StorageClusterList struct {
+	meta.TypeMeta `json:",inline"`
+	meta.ListMeta `json:"metadata,omitempty"`
+	Items         []StorageCluster `json:"items"`
+}
+
+// StorageClusterSpec is the spec used to define a storage cluster
+type StorageClusterSpec struct {
+	// Image is docker image of the storage driver
+	Image string `json:"image"`
+	// Kvdb is the information of kvdb that storage driver uses
+	Kvdb KvdbSpec `json:"kvdb"`
+	// Network is the network information for storage driver
+	Network NetworkSpec `json:"network"`
+	// Storage details of storage used by the driver
+	Storage StorageSpec `json:"storage"`
+	// CloudStorage details of storage in cloud environment.
+	// Keep this only at cluster level for now until support to change
+	// cloud storage at level is added.
+	CloudStorage CloudStorageSpec `json:"cloudStorage"`
+	// SecretsProvider is the name of secret provider that driver will connect to
+	SecretsProvider string `json:"secretsProvider"`
+	// Env is a list of environment variables used by the driver
+	Env []core.EnvVar `json:"env"`
+	// CSIEndpoint is the csi endpoint for the driver
+	CSIEndpoint string `json:"csiEndpoint"`
+	// CallHome send cluster information for analytics
+	CallHome bool `json:"callHome"`
+	// RuntimeOpts is a map of options with extra configs for storage driver
+	RuntimeOpts map[string]string `json:"runtimeOptions"`
+	// Nodes node level configurations that will override the ones at cluster
+	// level. These configurations can be grouped based on label selectors.
+	Nodes []NodeSpec `json:"nodes"`
```

Comment:
This is missing a placement spec that we will use for the daemonset. By default, we use the following that install.portworx.com returns. But we will need users to override that else we get stuck on an install which requires custom placement.

```
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: px/enabled
                operator: NotIn
                values:
                - "false"
              - key: node-role.kubernetes.io/master
                operator: DoesNotExist
```

https://rook.io/docs/rook/v0.5/cluster-crd.html#placement-configuration-settings is a good reference for placement spec.